.vitepress/theme/components/Feedback.vue

```diff
@@ -0,0 +1,208 @@
+<script setup lang="ts">
+import { watch, ref } from 'vue';
+import { useRoute } from 'vitepress';
+import LoadingIcon from './LoadingIcon.vue';
+import { sendEvent } from '../scripts/gtag';
+
+enum FeedbackState {
+  START = 'START',
+  YES = 'YES',
+  NO = 'NO',
+  END = 'END',
+}
+
+const route = useRoute();
+const currentState = ref(FeedbackState.START);
+const feedback = ref('');
+const isSending = ref(false);
+const hasError = ref(false);
+
+function onButtonClick(value: FeedbackState.YES | FeedbackState.NO) {
+  currentState.value = value;
+  sendEvent({
+    eventName: 'feedback_helpful',
+    pagePath: route.path,
+    value,
+  });
+}
+
+function encode(data: Record<string, string>) {
+  return Object.keys(data)
+    .map((key) => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
+    .join('&');
+}
+
+// We send a form to Netlify: https://docs.netlify.com/forms/setup/
```

Comment:
❓ I'm not sure if I set up a form correctly and how I can test it. Right now it returns 404 when I submit it but I'm not sure if it's supposed to work in preview 🤔